src/utils.py

```diff
@@ -1515,6 +1515,75 @@ def run_hill_climbing(
     return states, actions, heuristics
 
 
+def run_policy_guided_astar(
+        initial_state: _S,
+        check_goal: Callable[[_S], bool],
+        get_valid_actions: Callable[[_S], Iterator[Tuple[_A, float]]],
+        get_next_state: Callable[[_S, _A], _S],
+        heuristic: Callable[[_S], float],
+        policy: Callable[[_S], Optional[_A]],
+        num_rollout_steps: int,
```

Comment:
what will this usually be in practice?

Reply:
we usually use 50 but it doesn't matter that much